include/maths/CDataFrameAnalysisInstrumentationInterface.h

```diff
@@ -50,16 +54,96 @@ class MATHS_EXPORT CDataFrameAnalysisInstrumentationInterface {
     }
     //! Factory for the nextStep() callback function object.
     TStepCallback stepCallback() {
-        return [this](std::uint32_t step) { this->nextStep(step); };
+        return [this](const std::string& phase) { this->nextStep(phase); };
     }
 };
 
-//! \brief Dummies out all instrumentation.
-class MATHS_EXPORT CDataFrameAnalysisInstrumentationStub final
-    : public CDataFrameAnalysisInstrumentationInterface {
+class MATHS_EXPORT CDataFrameOutliersInstrumentationInterface
+    : virtual public CDataFrameAnalysisInstrumentationInterface {};
+
+//! \brief Instrumentation interface for Supervised Learning jobs.
+//!
+//! DESCRIPTION:\n
+//! This interface extends CDataFrameAnalysisInstrumentationInterface with a setters
+//! for hyperparameters, validatioin loss results, and job timing.
+class MATHS_EXPORT CDataFrameTrainBoostedTreeInstrumentationInterface
+    : virtual public CDataFrameAnalysisInstrumentationInterface {
+public:
+    enum EStatsType { E_Regression, E_Classification };
+    struct SRegularization {
+        SRegularization() = default;
+        SRegularization(double depthPenaltyMultiplier,
+                        double softTreeDepthLimit,
+                        double softTreeDepthTolerance,
+                        double treeSizePenaltyMultiplier,
+                        double leafWeightPenaltyMultiplier)
+            : s_DepthPenaltyMultiplier{depthPenaltyMultiplier},
+              s_SoftTreeDepthLimit{softTreeDepthLimit}, s_SoftTreeDepthTolerance{softTreeDepthTolerance},
+              s_TreeSizePenaltyMultiplier{treeSizePenaltyMultiplier},
+              s_LeafWeightPenaltyMultiplier{leafWeightPenaltyMultiplier} {};
+        double s_DepthPenaltyMultiplier = 0.0;
+        double s_SoftTreeDepthLimit = 0.0;
+        double s_SoftTreeDepthTolerance = 0.0;
+        double s_TreeSizePenaltyMultiplier = 0.0;
+        double s_LeafWeightPenaltyMultiplier = 0.0;
+    };
+    struct SHyperparameters {
+        double s_Eta = 0.1;
```

Comment:
I would initialise all the values in this class to something very obviously wrong: if we miss overwriting them that's a bug and we want to be able to pick that up easily when we review the output, e.g.
```suggestion
        double s_Eta = -1.0;
```